### PR TITLE
Smp

### PR DESCRIPTION
**Pull request by lyakh**
Unify core counting.

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
Thanks for fixing this, not sure why we ever had two macros for core count. Should this be on top of your latest Zephyr PR or is it good for master ?

**Comment by lyakh** [comment]
> Thanks for fixing this, not sure why we ever had two macros for core count. Should this be on top of your latest Zephyr PR or is it good for master ?

@lgirdwood it should be good for master, yes. If / when this gets merged, then https://github.com/thesofproject/sof/pull/3502/commits/f03de8c751e1fca6d05187fbad098a2a42fd0631 will simplify a bit. So, once we decide that yes, we want to take these, I'll update that PR.

File: src/arch/xtensa/Kconfig [hunk 1]
```diff
@@ -4,10 +4,17 @@
 
 menu "Xtensa Architecture"
 
+config MAX_CORE_COUNT
+	int
+	default 1 if KERNEL_BIN_NAME = "zephyr" || LIBRARY || IMX8 || IMX8X || \
+		BAYTRAIL || HASWELL || CHERRYTRAIL || BROADWELL || HOST
```
Comment (paulstelian97) [inline src/arch/xtensa/Kconfig:10]: No IMX8M?

Comment (lyakh) [reply src/arch/xtensa/Kconfig]: fixed, thanks

**Review by paulstelian97 — APPROVED** [review]
Barring the PR title (which won't show up in the git log anyway) this looks good from my point of view.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
I've also tagged this for v1.6 since it's fixing an issue.

**Pull request merged**